x-pack/plugins/lens/public/indexpattern_datasource/dimension_panel/operation_support.ts

```diff
@@ -61,3 +63,25 @@ export const getOperationSupportMatrix = (props: Props): OperationSupportMatrix
     fieldByOperation: supportedFieldsByOperation,
   };
 };
+
+export const getOperationsForField = (
+  field: IndexPatternField,
+  filterOperations: (operation: OperationMetadata) => boolean
+): Set<string> | undefined => {
+  const supportedOperations: Set<OperationType> = new Set();
+  operationDefinitions
+    .sort(getSortScoreByPriority)
```

Comment:
What's the reason for sorting here?
It seems not required from the code, but maybe I missing some bits.

Reply:
It's there so when you drop a field in the empty dimension it will convert to an operation with highest priority, so for example for metric operations it will be always Average for number fields.

Reply:
This function returns a `Set` of strings in no particular order. I see no value in sorting it before creating the `Set`, unless the extracted `keys` require to be in insertion order. But from the code flow it seems that the `Set` is only used for lookups.

Reply:
🤔 Where do you see it used as a lookup? It's actually the opposite, it's used as an iterator (to get the first element), eg here: https://github.com/elastic/kibana/pull/90744/files?file-filters%5B%5D=.ts#diff-3e9e8b7caacb2745a55324a131a0823d007bd324ee4acb4a1eead81a1d96f07bR190

Your comments made me think array is much better suited here. Refactored in my last commit!